### ISSUE TITLE
Table of contents element not 508/WCAG 2.1 AA compliant

### ISSUE THREAD

**Issue by rnewhook586** [elementor/elementor#23879]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Multiple issues:
Not navigable via keyboard/screen-reader
Controlled by two small divs instead of making whole area selectable.
ARIA attributes are not used & does not use plain language to describe state.

EXERPT FROM FIXES:
The below adds aria labels, event listeners seem to not work due to how this was built out so we really need this fixed at the code-level. This is to address multiple [recrements](https://www.w3.org/TR/WCAG21). Additionally, these to not address the issue of throwing focus once a section is selected which fulfills areas in 3.2. Optimally the focus will be thrown to the relevant section header once selected, and the TOC would collapse.

   `   // Fixes for base accessibility
            elementorTOC_Chevron.setAttribute("tabindex", "0");
            elementorTOC_Chevron.setAttribute("role", "button");
            elementorTOC_Chevron.setAttribute("aria-expanded", "false");
            elementorTOC_Chevron.setAttribute("aria-expanded", "collapsed table of contents");
            elementorTOC_Chevron.setAttribute("aria-label", "table of contents");

// What to do when selected
                
               //Open the chevron. Open Chevron div is unhidden, hide the closed one.
                function open_elementorTOC_Chevron() {
                    document.querySelector('.elementor-toc__toggle-button--expand').click();
                    state_elementorTOC_Chevron=1;
                    elementorTOC_Chevron.setAttribute("aria-expanded", "true");
                    console.log("O - Open TOC");
                };
                      
                //Close the chevron. Closed Chevron div is unhidden, hide the open one.
                function close_elementorTOC_Chevron() {
                    document.querySelector('.elementor-toc__toggle-button--collapse').click();
                    state_elementorTOC_Chevron=0;
                    elementorTOC_Chevron.setAttribute("aria-expanded", "false");
                     console.log("X - Close TOC");
                };`

### Steps to reproduce

Use keyboard only to navigate, use screen-reader. The issue is present even with accessibility optimizations toggled on within Elementor.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

Affects all browsers. Tested on Mac OS using reader utility, keyboard, & inspector - this is a codebase issue.

**Comment by rnewhook586** [comment]
Updated post to include sys info.

**Comment by askwpgirl** [comment]
@nicholaszein - When will this be addressed? Aside from this GitHub report, this issue has been shared numerous times with the development team long before this. I have clients who use TOC widget, and we are looking now at pulling it because it is not accessible.

**Comment by askwpgirl** [comment]
Any update on this issue @rami-elementor ? As this report is coming up on its one-year anniversary, do you have any code snippets to share until this is fixed in core? Passing accessibility is very important from a legal standpoint not to mention in this case for users who cannot use a mouse for whatever reason. I am looking into third-party Elementor TOC widgets I can recommend in the meantime. 

Elementor has a few widgets that are navigation in nature that fail accessibility. Since navigation of a website is a crucial part of accessibility, I hope we can bring some attention to this soon. Thanks! Let me know how I can help.
